packages/ace-mode/index.js

```diff
@@ -269,942 +26,89 @@ ace.define(
   [
     'require',
     'exports',
     'module',
     'ace/lib/oop',
-    'ace/mode/text',
-    'ace/mode/mongodb_highlight_rules',
-    'ace/mode/matching_brace_outdent',
-    'ace/worker/worker_client',
-    'ace/mode/behaviour/cstyle',
-    'ace/mode/folding/cstyle'
+    'ace/mode/javascript',
+    'ace/mode/mongodb_highlight_rules'
   ],
-  function(acequire, exports, module) {
-    'use strict';
-
-    var oop = acequire('../lib/oop');
-    var TextMode = acequire('./text').Mode;
-    var MongoDBHighlightRules = acequire('./mongodb_highlight_rules')
-      .MongoDBHighlightRules;
-    var MatchingBraceOutdent = acequire('./matching_brace_outdent')
-      .MatchingBraceOutdent;
-    var WorkerClient = acequire('../worker/worker_client').WorkerClient;
-    var CstyleBehaviour = acequire('./behaviour/cstyle').CstyleBehaviour;
-    var CStyleFoldMode = acequire('./folding/cstyle').FoldMode;
-
-    var Mode = function() {
+  function (acequire, exports) {
+    const oop = acequire('../lib/oop');
+    const { Mode: JavaScriptMode } = acequire('./javascript');
+    const { MongoDBHighlightRules } = acequire('./mongodb_highlight_rules');
+    // MongoDB mode is just javascript mode with a slightly modified highligher
+    // (see highligher code above for details)
+    function MongoDBMode(...args) {
+      oop.mixin(this, new JavaScriptMode(...args));
       this.HighlightRules = MongoDBHighlightRules;
-
-      this.$outdent = new MatchingBraceOutdent();
-      this.$behaviour = new CstyleBehaviour();
-      this.foldingRules = new CStyleFoldMode();
-    };
-    oop.inherits(Mode, TextMode);
-
-    (function() {
-      this.lineCommentStart = '//';
-      this.blockComment = { start: '/*', end: '*/' };
-      this.$quotes = { '"': '"', "'": "'", '`': '`' };
-
-      this.getNextLineIndent = function(state, line, tab) {
-        var indent = this.$getIndent(line);
-
-        var tokenizedLine = this.getTokenizer().getLineTokens(line, state);
-        var tokens = tokenizedLine.tokens;
-        var endState = tokenizedLine.state;
-
-        if (tokens.length && tokens[tokens.length - 1].type == 'comment') {
-          return indent;
-        }
-
-        if (state == 'start' || state == 'no_regex') {
-          var match = line.match(/^.*(?:\bcase\b.*:|[\{\(\[])\s*$/);
-          if (match) {
-            indent += tab;
-          }
-        } else if (state == 'doc-start') {
-          if (endState == 'start' || endState == 'no_regex') {
-            return '';
-          }
-          var match = line.match(/^\s*(\/?)\*/);
-          if (match) {
-            if (match[1]) {
-              indent += ' ';
-            }
-            indent += '* ';
-          }
-        }
-
-        return indent;
+      // We completely disable javascript worker because the one packaged with
+      // javascript mode doesn't work correctly when parsing queries or
+      // aggregations showing confusing errors in the editor
+      this.createWorker = () => {
```

Comment:
This doesn't really change anything right now as workers are globally disabled for all editors, but will be important later. I tried keeping javascript worker and even though it works when loaded, the way it tries to process editor values that we have leads to absolutely messed up annotations visible in the editor